runner.py

```diff
@@ -0,0 +1,647 @@
+            config['type'] = 'harness'
+            # Create path to harness
+            config['harness'] = config['path'] / config['harness']
+            if not config['harness'].exists() or not str(config['harness']).endswith('.py'):
+                raise YAMLParserError(self, f"harness {config['harness']} file not found")
+
+        return config
+
+    def parse_test_config(self):
+        config = self.config
+        if 'test' not in config:
+            raise YAMLParserError(self, 'Keyword "test" not found in test config')
+
+        config = config['test']
+        if 'tests' not in config:
+            raise YAMLParserError(self, 'Keyword "tests" not found in test config')
+
+        for test_config in config['tests']:
+            # Test consist of one (key, value) which is (name, config)
+            if len(test_config.keys()) != 1:
+                raise YAMLParserError(self, "test name not found in test config")
+
+            name = list(test_config.keys())[0]
+            test_config = test_config[name]
+
+            # Add useful staff to config
+            name = remove_prefix(str(self.path), str(PRTOS_PROJECT_DIR) + '/') + '/' + name
+            test_config['name'] = name.replace('/', '.')
+            test_config['path'] = self.path
+
+            # Combine 'local' and 'global' config
+            for key in config:
+                if key not in test_config and key != 'tests':
+                    test_config[key] = config[key]
+
+            test = self.parse_test_case(test_config)
+            if test:
+                self.parsed_tests.append(test)
+
+        return self.parsed_tests
+
+
+def remove_prefix(string, prefix):
+    if string.startswith(prefix):
+        return string[len(prefix):]
+
+    return string
+
+
+def search_for_tests(paths=None):
+    if not paths:
+        test_paths = list(PRTOS_TEST_DIR.rglob('test*.yaml'))
+    else:
+        test_paths = []
+        for path in paths:
+            candidate = list(path.rglob('test*.yaml'))
+            if not candidate:
+                raise ValueError(f'Test {path} does not contain .yaml test configuration')
+            test_paths.extend(candidate)
+
+    logging.debug("Following test configs has been found:")
+    for path in test_paths:
+        logging.debug(path)
+
+    return test_paths
+
+
+def set_logger(level=logging.INFO):
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setLevel(level)
+    formatter = logging.Formatter('%(message)s')
+    stream_handler.setFormatter(formatter)
+    root.addHandler(stream_handler)
+
+
+def args_log_level(arg):
+    levels = {'DEBUG': 1, 'INFO': 2, 'WARNING': 3, 'ERROR': 4}
+    arg = levels.get(arg)
+
+    if not arg or arg not in (1, 2, 3, 4):
+        logging.warning(f"{arg} is not valid verbosity level")
+        sys.exit(1)
+
+    # Convert to logging module level (10, 20, 30, 40)
+    return arg * 10
```

Comment:
```suggestion
    levels = {'DEBUG': 1, 'INFO': 2, 'WARNING': 3, 'ERROR': 4}
    arg1 = levels.get(arg)

    if not arg1 or arg1 not in (1, 2, 3, 4):
        logging.warning(f"{arg} is not valid verbosity level")
        sys.exit(1)

    # Convert to logging module level (10, 20, 30, 40)
    return arg1 * 10
```
Now if  `arg` does not contain a valid verbosity level a confusing message is displayed:
`WARNING:root:None is not valid verbosity level`